From 73473e64057b7f80fc3fa4a9c2379ba9acbf2e16 Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Fri, 28 Jan 2022 19:47:53 +0200
Subject: [PATCH] [SW-65638] gaudi2: release: enhance the ignore

Sometimes there is a need to ignore at least one o fhte system entities,
such as the JC/bmc-host, HiB/bmc-oam or the host itself.
For now there no need to allow ignoring the HiB.

instead of having multiple flags with long names, it was consolidate
under a single instruction with arguments (-I)

Change-Id: Icfd69d7f91b3d83ba0176e4ef98031da9c64c6aa
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/scripts/fwi/fw_installer.sh b/scripts/fwi/fw_installer.sh
index 8287278..444554a 100755
--- a/scripts/fwi/fw_installer.sh
+++ b/scripts/fwi/fw_installer.sh
@@ -112,7 +112,7 @@
 local_virt_file=~/my-virt-xml
 
 ssh_rmt_user_args=
-
+ignore_args=()
 devs_list=()
 user_devs_list=()
 devs_list_pci_ids=()
@@ -229,7 +229,10 @@
 	--oam-pwr <on|off|cycle>  : power cycle only the OAMs, may need a host reboot, it may
 	                            cause a host reboot.
 	--rmt-host-reboot         : exlicitly send a request to reboot the remote host.
-	-I,--ignore-rmt-host      : in case remote host is not relevant to the operation such
+	-I|--ignore [h|host][jc|bh|bmc-host]
+	                          : explicitly ignores one or more entities in the setup. such
+	                            with no arguent ignores only the host.
+	                            EX, in case remote host is not relevant to the operation such
                                     as toggling WP, ignores the absent of the rmt-host
 	--pre-reboot              : do a reboot to the system before the update procedure
 	-l|--login <host|bmc>     : open an ssh on the host or the OAM's bmc
@@ -999,6 +1002,28 @@
 	return 0
 }
 
+function handle_ignore_op()
+{
+	[ ${#ignore_args[@]} -eq 0 ] && ignore_args=(host)
+
+	for elem in ${ignore_args[@]}; do
+		case $elem in
+		host|h)
+			rmt_host_ignore_f=1
+			skip_copy_f=1
+			;;
+		hib|bmc|bo|ob|bmc-oam|oam-bmc)
+			bmc_oam_ignore_f=1
+			;;
+		jc|bh|hb|bmc-host|bhost)
+			bmc_host_ignore_f=1
+			;;
+		*)
+			;;
+		esac
+	done
+}
+
 function get_user_params()
 {
 	local err
@@ -1084,6 +1109,9 @@
 			;;
 		-k|--ignore-ssh-init)
 			ignore_ssh_init_f=1
+			;;
+		-I)
+			ignore_args=(${args[@]})
 			;;
 		-I|--ignore-rmt-host)
 			rmt_host_ignore_f=1
@@ -1759,6 +1787,7 @@
 	create_config_file_op
 	api_sanity_check
 	[ $? -ne 0 ] && return 1
+	handle_ignore_op
 	run_stages
 	pinfo "For more details see logs directory ${log_dir}/"
 }
